src/utils.c

```diff
@@ -363,3 +363,84 @@ SEXP coerceUtf8IfNeeded(SEXP x) {
   return(ans);
 }
 
+// if (length(x)) length(x[[1L]]) else 0L
+int NROW(SEXP x) {
+  if (!LENGTH(x))
+    return 0; // # nocov # not yet reached from anywhere, cbindlist uses it but escapes for !NCOL(x)
+  return length(VECTOR_ELT(x, 0));
+}
+
+// length(x)
+int NCOL(SEXP x) {
```

Comment:
i don't see the point of this? there's already `LENGTH` and you even describe it in the comment as `length(x)`

Reply:
just for consistency to having `NROW` function, to abstract list type, as we do for `NROW`.

Reply:
Can we do this as a `#define` macro?

Reply:
it is unlikely to be called in a big loops, so macro might not be that much important here. And function we could eventually export.